appinventor/appengine/src/com/google/appinventor/client/youngandroid/YoungAndroidFormUpgrader.java

```diff
@@ -511,6 +511,19 @@ private static int upgradeBarcodeScannerProperties(Map<String, JSONValue> compon
       // The UseExternalScanner property was added.
       srcCompVersion = 2;
     }
+    if (srcCompVersion < 3) {
+      // The UseExternalScanner property was set to True by default.
+      if (componentProperties.containsKey("UseExternalScanner")){
+        String value = ((ClientJsonString)componentProperties.get("UseExternalScanner")).getString();
+        if (value.equals("True")){
+          componentProperties.remove("UseExternalScanner");
+        }
+      }
+      else {
+        componentProperties.put("UseExternalScanner", new ClientJsonString("False"));
```

Comment:
Likewise, if the key isn't there, then it was the previous default (True), so you should be setting this to True here, not False.